keras/engine/topology.py

```diff
@@ -806,33 +856,78 @@ def output_shape(self):
                             'ill-defined for the layer. ' +
                             'Use `get_output_shape_at(node_index)` instead.')
 
-    def add_updates(self, updates, inputs):
+    def add_loss(self, losses, inputs=None):
+        if losses is None:
+            return
+        # Update self.losses
+        losses = to_list(losses)
+        if not hasattr(self, 'losses'):
+            self.losses = []
+        try:
```

Comment:
We don't actually want to skip the case "is property", because we may well have a property with a getter, in which case appending to list would be the right behavior (e.g. this how `trainable_weights` currently work).

We explicitly want to skip the case "not settable", distinct from "is property". It isn't clear what is the best way to check if an attribute is settable, by try/except seems like a good candidate: http://stackoverflow.com/questions/6102747/checking-if-property-is-settable-deletable-in-python